dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java

```diff
@@ -149,14 +153,16 @@ public void onGoaway(Channel channel) {
 
     public void onConnected(Channel channel) {
         this.channel.set(channel);
+        this.onConnectFuture.complete(new Object());
         channel.attr(CONNECTION).set(this);
         if (logger.isInfoEnabled()) {
             logger.info(String.format("%s connected ", this));
         }
     }
 
-    public void connectSync() {
+    public void connectSync() throws InterruptedException, ExecutionException, TimeoutException {
         this.initPromise.awaitUninterruptibly(this.connectTimeout);
+        this.onConnectFuture.get(this.connectTimeout, TimeUnit.MILLISECONDS);
```

Comment:
Here use CountDownLatch will be better.

Reply:
And here onConnectFuture is stateful, it should be reset when connect close.

Reply:
yes,i will update it today

Reply:
Maybe say sry jiangxch, this pr is duplicated with #8126, thanks for your contribution.

Reply:
ok